webapp/tests/forms/flows/test_step_chooser.py

```diff
@@ -101,6 +128,124 @@ def test_update_data_is_not_called_if_update_data_not_set(self):
+        assert returned_prev_step is None
+
+
+class TestDetermineNextStep:
+    @pytest.fixture
+    def step_chooser_without_preconditions(self):
+        testing_steps = [MockStartStep, MockFormWithInputStep, MockMiddleStep, MockFinalStep]
+        yield StepChooser(title="Testing StepChooser", steps=testing_steps,
+                          endpoint="lotse", overview_step=MockFormWithInputStep)
+
+    @pytest.fixture
+    def step_chooser_with_preconditions(self):
+        testing_steps = [MockStartStep, MockFormWithInputStep, MockStepWithPrecondition, MockFinalStep]
+        yield StepChooser(title="Testing StepChooser", steps=testing_steps,
+                          endpoint="lotse", overview_step=MockFormWithInputStep)
+
+    def test_if_next_steps_have_no_precondition_return_direct_successor(self, step_chooser_without_preconditions):
+        expected_next_step = MockMiddleStep
+        actual_next_step = step_chooser_without_preconditions.determine_next_step(MockFormWithInputStep.name, {})
+        assert actual_next_step == expected_next_step
+
+    def test_if_next_steps_have_no_precondition_request_only_direct_successor(self, step_chooser_without_preconditions):
+        with patch('tests.forms.mock_steuerlotse_steps.MockStartStep.check_precondition') as start_step_check, \
+             patch('tests.forms.mock_steuerlotse_steps.MockFormWithInputStep.check_precondition') as input_step_check, \
+             patch('tests.forms.mock_steuerlotse_steps.MockMiddleStep.check_precondition') as middle_step_check, \
+             patch('tests.forms.mock_steuerlotse_steps.MockFinalStep.check_precondition') as final_step_check:
+            step_chooser_without_preconditions.determine_next_step(MockFormWithInputStep.name, {})
+            start_step_check.assert_not_called()
+            input_step_check.assert_not_called()
+            middle_step_check.assert_called_once()
+            final_step_check.assert_not_called()
+
+    def test_if_next_steps_have_met_precondition_return_direct_successor(self, step_chooser_with_preconditions):
+        expected_next_step = MockStepWithPrecondition
+        actual_next_step = step_chooser_with_preconditions.determine_next_step(MockFormWithInputStep.name,
+                                                                               {'precondition_met': True})
+        assert actual_next_step == expected_next_step
+
+    def test_if_next_steps_have_but_dont_meet_precondition_return_step_after(self, step_chooser_with_preconditions):
+        expected_next_step = MockFinalStep
+        actual_next_step = step_chooser_with_preconditions.determine_next_step(MockFormWithInputStep.name,
+                                                                               {'precondition_met': False})
+        assert actual_next_step == expected_next_step
+
+    def test_if_next_steps_have_but_dont_meet_precondition_request_correct_steps(self, step_chooser_with_preconditions):
```

Comment:
Maybe we can quickly talk together about that test?

Reply:
rename to `request_direct_successors`